src/VisualStudio/Core/SolutionExplorerShim/AnalyzerItemTracker.cs

```diff
@@ -17,32 +17,39 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplore
     [Export]
     internal class AnalyzerItemsTracker : IVsSelectionEvents
     {
+        private IServiceProvider _serviceProvider;
         private IVsMonitorSelection _vsMonitorSelection = null;
         private uint _selectionEventsCookie = 0;
 
         public event EventHandler SelectedHierarchyItemChanged;
 
         [ImportingConstructor]
-        public AnalyzerItemsTracker(SVsServiceProvider serviceProvider)
+        public AnalyzerItemsTracker(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
         {
-            _vsMonitorSelection = serviceProvider.GetService(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
+            _serviceProvider = serviceProvider;
         }
 
         public void Register()
         {
```

Comment:
We need to avoid loading other DLLs if at all possible, so I don't want to take a dependency on a type in another assembly. Even without this change, this code would be badly broken if we were being called off the foreground thread.